web/src/hooks/useDAGChart.ts

```diff
@@ -1,8 +1,14 @@
 import * as d3DAG from 'd3-dag';
+import {Dag} from 'd3-dag';
 import _ from 'lodash';
-import { TSpanMap } from '../components/Diagram/components/DAG';
+import {TSpanMap} from '../components/Diagram/components/DAG';
 
-export const useDAGChart = (spanMap: TSpanMap = {}) => {
+export const useDAGChart = (
+  spanMap: TSpanMap = {}
+): void | {
+  dag: Dag<{id: string; parentIds: string[]}, undefined>;
```

Comment:
Should we encapsulate this into an interface?

Reply:
sure